misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/FunctionInvocationExpressionEvaluator.java

```diff
@@ -83,69 +87,39 @@ public BExpressionValue evaluate() throws EvaluationException {
         }
     }
 
-    /**
-     * Searches for a matching jvm method for a given ballerina function using its syntax node and the debug context
-     * information.
-     *
-     * @return the matching JVM method, if available
-     */
-    private Optional<GeneratedStaticMethod> findFunctionFromLoadedClasses() {
-        List<ReferenceType> allClasses = context.getAttachedVm().allClasses();
-        DebugSourceType sourceType = context.getSourceType();
-        for (ReferenceType cls : allClasses) {
-            try {
-                // Expected class name should end with the file name of the ballerina source, only for single
-                // ballerina sources. (We cannot be sure about the module context, as we can invoke any method
-                // defined within the module.)
-                if (sourceType == DebugSourceType.SINGLE_FILE && !cls.name().endsWith(context.getFileName().get())) {
-                    continue;
-                }
-                // If the sources reside inside a ballerina module/project, generated class name should start with the
-                // organization name of the ballerina module/project source.
-                if (sourceType == DebugSourceType.PACKAGE && !cls.name().startsWith(context.getPackageOrg().get())) {
-                    continue;
-                }
-                List<Method> methods = cls.methodsByName(functionName);
-                for (Method method : methods) {
-                    // Note - All the ballerina functions are represented as java static methods and all the generated
-                    // jvm methods contain strand as its first argument.
-                    if (method.isStatic()) {
-                        return Optional.of(new GeneratedStaticMethod(context, cls, method));
-                    }
-                }
-            } catch (ClassNotPreparedException ignored) {
-                // Unprepared classes should be skipped.
-            }
+    private Optional<FunctionSymbol> findFunctionWithinModule() {
+        SemanticModel semanticContext = context.getDebugCompiler().getSemanticInfo();
+        LinePosition position = LinePosition.from(context.getLineNumber(), 0);
+        List<Symbol> functionMatches = semanticContext.visibleSymbols(context.getFileNameWithExt().get(), position)
```

Comment:
If the goal is to find the function from within the constructs in the module itself, you can use the `moduleLevelSymbols()` API.

Reply:
+1. Changed in e1c2ab6